apps/apps.c

```diff
@@ -423,7 +426,16 @@ static char *app_get_pass(const char *arg, int keepbio)
                 return NULL;
             }
         } else {
-            BIO_printf(bio_err, "Invalid password argument \"%s\"\n", arg);
+            /* argument syntax error; do not reveal too much about arg */
+            tmp = strchr(arg, ':');
+            if (tmp == NULL || tmp - arg > PASS_SOURCE_SIZE_MAX)
+                BIO_printf(bio_err,
+                           "Invalid password argument, missing ':' within the first %d chars\n",
+                           PASS_SOURCE_SIZE_MAX + 1);
+            else
+                BIO_printf(bio_err,
+                          "Invalid password argument, starting with \"%.*s\"\n",
+                           (int)(tmp - arg + 1), arg);
```

Comment:
I would actually make it much simpler:

``` C
            BIO_printf(bio_err, "Invalid password argument\n");
```

Reply:
This would indeed be much simpler but also much less helpful in case of syntax errors.
In particular, most users are not aware that they should prepend verbatim password input with `pass:`.

Reply:
So for the sake of simplicity we could leave out the (partial) echoing of the password argument. What do others think on that?

And for helpfulness, how about this?
```
 BIO_printf(bio_err, "Invalid password argument. Verbatim input must be preceded with 'pass:'\n");
```
which would give a hint regarding the most common mistake using the `-pass` option, reducing the need for questions like https://superuser.com/questions/724986/how-to-use-password-argument-in-via-command-line-to-openssl-for-decryption